packages/jellyfish-transaction/src/buffer/buffer_composer.ts

```diff
@@ -209,6 +209,32 @@ export abstract class ComposableBuffer<T> implements BufferComposer {
     }
   }
 
+  /**
+   *  HEX String without length specified, encoded into Buffer as the same order of the Hex String.
+   *  In short this read a hex and push it into the Buffer. It will not re-order the endian.
+   * `toBuffer` defaults to '' and writes 00 to buffer
+   *
+   * @param getter to read HEX String. Writes its length then write the HEX string. Defaults to empty string.
+   * @param setter to read ordered Buffer and set as the same ordered HEX String
+   */
+  static optionalVarUIntHex (getter: () => string | undefined, setter: (data: string | undefined) => void): BufferComposer {
```

Comment:
`optionalVarHex` may be?
two properties, optionality and variability

Reply:
I combined optionalHex and varUIntUtf8BE to name it.
Without the UInt, I think we lose information about it writing the length as VarUInt before the hex string